src/Library/demos/Event Controllers/main.vala

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env -S vala workbench.vala --pkg libadwaita-1
+
+private bool ctrl_pressed = false;
+
+public void main () {
+  // Gtk.Window hides Gtk.Widget's add_controller method, thus we need to access it as a Widget.
+  Gtk.Widget window = workbench.window;
```

Comment:
use `Gtk.Window` and cast it to `Gtk.Widget` when calling the add_controller method